src/models/user.py

```diff
@@ -0,0 +1,88 @@
+import bcrypt
+import sqlalchemy as sa
+from sanic import Sanic
+from sqlalchemy.future import select
+from sqlalchemy.orm import validates
+
+from src.models import Base
+
+
+class User(Base):
+    """User model."""
+
+    __tablename__ = "users"
+
+    id = sa.Column(sa.Integer, primary_key=True)
+
+    username = sa.Column(sa.String(255), nullable=False, unique=True)
+    password_hash = sa.Column(sa.String(255), nullable=False)
+
+    __mapper_args__ = {"eager_defaults": True}
+
+    @validates("username")
+    def validate_username(self, key: str, username: str) -> str:
+        """Validate username."""
+        assert len(username) > 1 and len(username) < 255
+        return username
+
+    @validates("password")
+    def validate_password(self, key: str, password: str) -> str:
+        """Validate password."""
+        assert len(password) > 1 and len(password) < 255
+        return password
+
+    def set_password(self, password: str) -> None:
+        """Calculates password hash and assigns it.
+
+        Args:
+            password (str): password string
+        """
+        self.password_hash = bcrypt.hashpw(
+            password.encode("utf-8"), bcrypt.gensalt()
+        ).decode("utf-8")
+
+    def check_password(self, password: str) -> bool:
+        """Compares `password` hash with current `password_hash`.
+
+        Args:
+            password (str): password string
+
+        Returns:
+            bool: whether the password matches the hash.
+        """
+        return bcrypt.checkpw(
+            password.encode("utf-8"), self.password_hash.encode("utf-8")
+        )
+
+    @staticmethod
+    async def get_user(app: Sanic) -> "User":
```

Comment:
Would it make sense to pass a `sqlalchemy` session as opposed to a `sanic` application?  It may result in a looser coupling and allow for easier testing.  Just a thought, not sure what is best.  Whatever you decide would also apply to the `create_user` method below.

Reply:
Let's see what we need when we write the tests and modify it accordingly. I used `app` to have more context in case it is needed (e.g. we could use `app.config` vars here).

Reply:
Yep makes sense 👍